lib/fs.js

```diff
@@ -283,35 +190,41 @@ fs.access = function(path, mode, callback) {
     throw new errors.TypeError('ERR_INVALID_CALLBACK');
   }
 
-  if (handleError((path = getPathFromURL(path)), callback))
-    return;
+  handleError(path = getPathFromURL(path));
+  nullCheck(path);
 
-  if (!nullCheck(path, callback))
-    return;
+  if (typeof path !== 'string' && !Buffer.isBuffer(path)) {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'path',
+                               ['string', 'Buffer', 'URL']);
+  }
+  if (!Number.isInteger(mode)) {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'mode', 'integer');
+  }
 
-  mode = mode | 0;
-  var req = new FSReqWrap();
+  const req = new FSReqWrap();
   req.oncomplete = makeCallback(callback);
   binding.access(pathModule._makeLong(path), mode, req);
 };
 
-fs.accessSync = function(path, mode) {
-  handleError((path = getPathFromURL(path)));
+fs.accessSync = function(path, mode = fs.F_OK) {
```

Comment:
As the validation of the mode is later it would be better performance wise to check for undefined (the `Number.isInteger` check is obsolete in that case). But this might be over optimizing stuff.